x-pack/plugins/spaces/server/new_platform/spaces_service/spaces_service.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { map, take } from 'rxjs/operators';
+import { Observable, Subscription, combineLatest } from 'rxjs';
+import { Legacy } from 'kibana';
+import { SavedObjectsService } from 'src/legacy/server/kbn_server';
+import {
+  Logger,
+  ElasticsearchServiceSetup,
+  HttpServiceSetup,
+  KibanaRequest,
+} from 'src/core/server';
+import { DEFAULT_SPACE_ID } from '../../../common/constants';
+import { SecurityPlugin } from '../../../../security';
+import { SpacesClient } from '../../lib/spaces_client';
+import { getSpaceIdFromPath } from '../../lib/spaces_url_parser';
+import { SpacesConfigType } from '../config';
+
+type RequestFacade = KibanaRequest | Legacy.Request;
+
+export interface SpacesServiceSetup {
+  scopedClient(request: RequestFacade): Promise<SpacesClient>;
+
+  getSpaceId(request: RequestFacade): string;
+
+  isInDefaultSpace(request: RequestFacade): boolean;
+}
+
+interface SpacesServiceDeps {
+  http: HttpServiceSetup;
+  elasticsearch: ElasticsearchServiceSetup;
+  savedObjects: SavedObjectsService;
+  getSecurity: () => SecurityPlugin | undefined;
+  config$: Observable<SpacesConfigType>;
+  spacesAuditLogger: any;
+}
+
+export class SpacesService {
+  private configSubscription$?: Subscription;
+
+  constructor(private readonly log: Logger, private readonly serverBasePath: string) {}
+
+  public async setup({
+    http,
+    elasticsearch,
+    savedObjects,
+    getSecurity,
+    config$,
+    spacesAuditLogger,
+  }: SpacesServiceDeps): Promise<SpacesServiceSetup> {
+    const getSpaceId = (request: RequestFacade) => {
+      const isLegacyRequest = typeof (request as any).getBasePath === 'function';
```

Comment:
`http.getBasePathFor` works for Legacy.Request. should we call it `isFakeRequest` and call only for FakeRequests?

Reply:
I'm happy to rename this to `isFakeRequest`. What do you mean by "only call for FakeRequests"? Is there a formal concept of a Fake Request outside of the reporting plugin?

Reply:
I found it in Upgrade assistance as well. https://github.com/restrry/kibana/blob/4b30212af64146e781d47aa29827fef0573b46da/x-pack/plugins/upgrade_assistant/server/lib/reindexing/worker.ts#L162 probably there are other places 🤷‍♂